spec/services/course_search_service_spec.rb

```diff
@@ -323,6 +323,24 @@
           expect(subject).to eq(expected_scope)
         end
       end
+
+      context 'when filter for undergraduate courses' do
```

Comment:
This whole file is 🤮

Reply:
I agree. This spec was SUPER difficult to grasp and understand. This spec needs a total rewrite...